src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,166 @@
+        if _upper_bound.value < _lower_bound.value:
+            raise ValueError(
+                (
+                    f"Illegal interval: Attempting to raise OutOfBoundsError, but given upper bound {_upper_bound} is "
+                    f"actually less than given lower bound {_lower_bound}."
+                ),
+            )
+        elif _lower_bound.check_lower_bound(actual) and _upper_bound.check_upper_bound(actual):
+            raise ValueError(
+                (
+                    f"Illegal interval: Attempting to raise OutOfBoundsError, but value {actual} is not actually"
+                    f" outside given interval {_lower_bound.str_lower_bound()}, {_upper_bound.str_upper_bound()}."
+                ),
+            )
+        super().__init__(f"{actual} is not inside {_lower_bound.str_lower_bound()}, {_upper_bound.str_upper_bound()}.")
+
+
+class Bound(ABC):
+    """Abstract base class for (lower or upper) Bounds on a float value."""
+
+    def __init__(self, value: float):
+        self._value = value
+
+    def __str__(self) -> str:
+        """Get a string representation of the concrete value of the Bound."""
+        return str(self.value)
+
+    @abstractmethod
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the Bound as the lower Bound of an interval."""
+
+    @abstractmethod
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the Bound as the upper Bound of an interval."""
+
+    @abstractmethod
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value does not exceed the Bound on the lower side."""
+
+    @abstractmethod
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value does not exceed the Bound on the upper side."""
+
+    @property
+    def value(self) -> float:
+        """Get the concrete value of the Bound."""
+        return self._value
+
+
+class ClosedBound(Bound):
+    """A closed Bound, i.e. the value on the border belongs to the range."""
+
+    def __init__(self, value: float):
+        super().__init__(value)
+
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the lower Bound of an interval."""
+        return f"[{self}"
+
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the upper Bound of an interval."""
+        return f"{self}]"
+
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value is not strictly lower than the ClosedBound."""
+        return value >= self._value
+
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value is not strictly higher than the ClosedBound."""
+        return value <= self._value
+
+
+class OpenBound(Bound):
+    """
+    An open Bound, i.e. the value on the border does not belong to the range.
+
+    May be infinite (unbounded).
```

Comment:
Better remove that since the way to specify that is to omit the bound rather than use the (now internal) bounds for infinity.

Reply:
Removed that.
However, the docstring of `Bound.__init__(self, value)` now states that *some* bounds *may* take allow `+/-inf` as a parameter.
I could also "forbid" it explicitely in the docstring for `Bound`, then still allow it in `OpenBound`, but that somehow feels incorrect.

Reply:
It's OK how you've done it. Alternatively, we could omit the discussion of infinity altogether for the bound classes since it's just meant for our internal use in the same file.

Reply:
I have changed the design to still allow creating `Bound`s with `float("inf")` (as we need to create these bounds internally), but to not allow passing such bounds to the constructor of the `OutOfBoundsError`. Upon trying, there will be a `ValueError` stating explicitely that `None` should be used.